src/components/boardComponents/useItemActions.js

```diff
@@ -125,6 +127,92 @@ export const useItemActions = () => {
     [getSelectedItemList, batchUpdateItems, selectedItems]
   );
 
+  // Align selection to a line
+  const alignAsLine = useRecoilCallback(
+    ({ snapshot }) => async () => {
+      const selectedItemList = await getSelectedItemList(snapshot);
+
+      // Compute
+      const minMax = { min: {}, max: {} };
+      minMax.min.x = Math.min(...selectedItemList.map(({ x }) => x));
+      minMax.min.y = Math.min(...selectedItemList.map(({ y }) => y));
+      minMax.max.x = Math.max(
+        ...selectedItemList.map(
+          ({ x, id }) => x + document.getElementById(id).clientWidth
+        )
+      );
+      minMax.max.y = Math.max(
+        ...selectedItemList.map(
+          ({ y, id }) => y //+ document.getElementById(id).clientHeight
+        )
+      );
+
+      const [newX, newY] = [minMax.min.x, minMax.max.y];
+      let index = -1;
+      batchUpdateItems(selectedItems, (item) => {
+        index += 1;
+        const { clientWidth, clientHeight } = document.getElementById(item.id);
+        return {
+          ...item,
+          x: newX + index * clientWidth,
+          y: newY - index,
+        };
+      });
+    },
+    [getSelectedItemList, batchUpdateItems, selectedItems]
+  );
+
+  // Align selection to an array
+  const alignAsSquare = useRecoilCallback(
+    ({ snapshot }) => async () => {
+      const selectedItemList = await getSelectedItemList(snapshot);
+
+      // Count number of elements
+      const numberOfElements = selectedItemList.length;
+      const numberOfColumns = Math.ceil(Math.sqrt(numberOfElements));
+
+      // Compute
+      const minMax = { min: {}, max: {} };
+      minMax.min.x = Math.min(...selectedItemList.map(({ x }) => x));
+      minMax.min.y = Math.min(...selectedItemList.map(({ y }) => y));
+      minMax.max.x = Math.max(
+        ...selectedItemList.map(
+          ({ x, id }) => x + document.getElementById(id).clientWidth
+        )
+      );
+      minMax.max.y = Math.max(
+        ...selectedItemList.map(
+          ({ y, id }) => y //+ document.getElementById(id).clientHeight
+        )
+      );
+
+      // const [newX, newY] = [
+      //   (minMax.min.x + minMax.max.x) / 2,
+      //   (minMax.min.y + minMax.max.y) / 2,
+      // ];
+      const [newX, newY] = [minMax.min.x, minMax.max.y];
+      let index = -1;
+      let currentColumn = -1;
+      let currentRow = 0;
+      batchUpdateItems(selectedItems, (item) => {
+        index += 1;
+        currentColumn += 1;
+        if (currentColumn + 1 > numberOfColumns) {
+          currentColumn = 0;
+          currentRow += 1;
+        }
+
+        const { clientWidth, clientHeight } = document.getElementById(item.id);
+        return {
+          ...item,
+          x: newX + currentColumn * clientWidth,
+          y: newY - currentRow * clientHeight,
```

Comment:
If found this is more standard to go from left to right (like you did) and from top to bottom (like reading). Have you any reason to do it from bottom to top ?

```suggestion
          y: newY + currentRow * clientHeight,
```

Reply:
I have no specific reason. Why not, to be defined on the use. And if necessary, be able to provide both directions (two distinct actions with two distinct icons).

Reply:
Fine !

Reply:
With this correction, we should no longer choose as the origin the image at the bottom left, but rather the image at the top left. I will make a correction.

Reply:
> 
> 
> With this correction, we should no longer choose as the origin the image at the bottom left, but rather the image at the top left. I will make a correction.

Also :

- Modify the icon.
- Place images in downward direction.

Reply:
Perfect!